temba/api/v2/views.py

```diff
@@ -4309,3 +4324,849 @@ def put(self, request, *args, **kwargs):
+                        "channel": "43cd6c9e-25cd-4512-bf29-d2999a4a27a3",
+                        "exclude": "Testers",
+                        "started_after": "2021-02-01",
+                        "started_before": "2021-03-13",
+                        "exited_after": "2021-02-01",
+                        "exited_before": "2021-03-13",
+                    }
+                )
+            ),
+        )
+
+
+class FlowVariableReportEndpoint(BaseAPIView):
+    """
+    This endpoint allows you to generate a report based on contact responses.
+
+    A **GET** returns groups split by results that contacts had responded
+
+    * **flow** - UUID of flow to which is need to prepare report
+    * **channel** - UUID or Name of channel to select only the contacts that received messages via that channel
+    * **exclude** - UUID or Name of contact group to select only contacts that not belong to that group
+    * **started_after** - Date, excludes all runs from the report that were started earlier a certain date
+    * **started_before** - Date, excludes all runs from the report that were started later a certain date
+    * **exited_after** - Date, excludes all runs from the report that were exited earlier a certain date
+    * **exited_before** - Date, excludes all runs from the report that were exited earlier a certain date
+    * **variables** - configuration which define the fields to be included in the report
+
+    Example:
+
+        POST /api/v2/flow_variable_report.json
+        {
+            "flow": "2f613ae3-2ed6-49c9-9161-fd868451fb6a",
+            "variables": {
+                "result_1": {
+                    "format": "value",
+                    "top": 3
+                }
+            }
+        }
+
+    Response:
+
+        {
+            "flow": "2f613ae3-2ed6-49c9-9161-fd868451fb6a",
+            "variables": {
+                "result_1": {
+                    "format": "value",
+                    "top": 3
+                }
+            },
+            "results": [
+                {
+                    "result_1": {
+                        "No": 1,
+                        "Yes": 1,
+                        "Other": 1
+                    }
+                }
+            ]
+        }
+
+    To generate report in CSV format pass query parameter 'export_csv':
+
+        POST /api/v2/flow_variable_report.json?export_csv=true
+    """
+
+    permission = "orgs.org_api"
+
+    @csv_response_wrapper
+    def post(self, request, *args, **kwargs):
+        applied_filters = {}
+        org = self.request.user.get_org()
+        try:
+            flow = Flow.objects.get(uuid=self.request.data.get("flow"))
+            applied_filters["flow"] = flow.uuid
+        except Flow.DoesNotExist:
+            return Response(
+                {"errors": {"flow": _("Please enter valid flow UUID.")}}, status=status.HTTP_400_BAD_REQUEST
+            )
+        queryset = FlowRun.objects.filter(org=org, flow=flow)
```

Comment:
Quick question: have you tested that on a large scale or should we ask the QA team to do that?

Reply:
I have only tested that on my local machine so we need QA to test it on a large scale.